Packs/CiscoFirepower/Integrations/CiscoFirepower/CiscoFirepower_test.py

```diff
@@ -316,3 +343,994 @@ def test_raw_response_to_context_access_policy(list_input, list_output):
+        'limit': '5',
+        'intrusion_policy_id': 'intrusion_policy_id',
+    }
+
+    with pytest.raises(ValueError) as ve:
+        from CiscoFirepower import list_intrusion_policy_command
+        list_intrusion_policy_command(mock_client, args)
+
+        assert str(ve) == 'GET and LIST arguments can not be supported simutanlesy.'
+
+
+def test_update_intrusion_policy_command(requests_mock, mock_client):
+    """
+    Scenario:
+    -   Update an intrusion policy.
+    Given:
+    -   intrusion_policy_id, name, basepolicy_id, description
+    When:
+    -    ciscofp-update-intrusion-policy is called.
+    Then:
+    -   Ensure the readable_output is correct.
+    -   Ensure the outputs_prefix is correct.
+    -   Ensure the outputs has no links.
+    """
+    args = {
+        'intrusion_policy_id': 'intrusion_policy_id',
+        'name': 'name',
+        'basepolicy_id': 'basepolicy_id',
+        'description': 'description',
+        'inspection_mode': 'PREVENTION'
+    }
+
+    method = 'PUT'
+    mock_response = load_mock_response('intrusion_policy_response.json')
+
+    requests_mock.request(
+        method,
+        f'{BASE_URL}/{SUFFIX}/policy/intrusionpolicies/{args["intrusion_policy_id"]}',
+        json=mock_response,
+    )
+
+    from CiscoFirepower import update_intrusion_policy_command
+    command_results = update_intrusion_policy_command(mock_client, args)
+
+    assert_command_results(
+        command_results=command_results,
+        method=method,
+        expected_output_prefix=INTRUSION_POLICY_CONTEXT
+    )
+
+
+@pytest.mark.parametrize(
+    'filename, status_code',
+    (
+        ('intrusion_policy_response.json', HTTPStatus.OK),
+        ('intrusion_policy_delete_fail.json', HTTPStatus.NOT_FOUND),
+    )
+)
+def test_delete_intrusion_policy_command(requests_mock, mock_client, filename, status_code):
+    """
+    Scenario:
+    -   Delete an intrusion policy.
+    -   Delete an intrusion policy that doesn't exist.
+    Given:
+    -   intrusion_policy_id
+    When:
+    -    ciscofp-delete-intrusion-policy is called.
+    Then:
+    -   Ensure the readable_output is correct.
+    """
+    args = {
+        'intrusion_policy_id': 'intrusion_policy_id'
+    }
+
+    method = 'DELETE'
+    mock_response = load_mock_response(filename)
+
+    requests_mock.request(
+        method,
+        f'{BASE_URL}/{SUFFIX}/policy/intrusionpolicies/{args["intrusion_policy_id"]}',
+        json=mock_response,
+        status_code=status_code
+    )
+
+    from CiscoFirepower import delete_intrusion_policy_command
+    command_results = delete_intrusion_policy_command(mock_client, args)
+
+    if status_code == HTTPStatus.OK:
+        assert_command_results(
+            command_results=command_results,
+            method=method
+        )
+
+    elif status_code == HTTPStatus.NOT_FOUND:
+        assert command_results.readable_output == \
+            f'The Intrusion Policy ID: "{args["intrusion_policy_id"]}" does not exist.'
```

Comment:
please split to two tests, here and in other similar cases.

Reply:
Done.